orttraining/orttraining/training_ops/cuda/math/softmax_grad.h

```diff
@@ -15,15 +15,21 @@ template <typename T>
 class SoftmaxGrad final : public CudaKernel {
  public:
   SoftmaxGrad(const OpKernelInfo& info) : CudaKernel{info} {
-    info.GetAttrOrDefault("axis", &axis_, static_cast<int64_t>(1));
-    log_softmax_ = info.GetKernelDef().OpName() == "LogSoftmaxGrad";
+    const auto& node = info.node();
+    opset_ = (node.OpType() == "SoftmaxGrad_13" || node.OpType() == "LogSoftmaxGrad_13") ? 13 : 1;
+    axis_ = info.GetAttrOrDefault("axis", static_cast<int64_t>(opset_ < 13 ? 1 : -1));
+    log_softmax_ = info.GetKernelDef().OpName() == "LogSoftmaxGrad" || info.GetKernelDef().OpName() == "LogSoftmaxGrad_13";
+    cuda_ep_ = const_cast<CUDAExecutionProvider*>(
+        static_cast<const CUDAExecutionProvider*>(info.GetExecutionProvider()));
   }
 
   Status ComputeInternal(OpKernelContext* context) const override;
 
  private:
   int64_t axis_;
   bool log_softmax_;
+  int opset_;  // opset_ of the forward Softmax/LogSoftmax operator
+  CUDAExecutionProvider* cuda_ep_;
```

Comment:
what we actually need is just he cuda_ep->GetDeviceProp(). 
We can cache this, rather than the EP pointer. 
const cudaDeviceProp& prop

Reply:
Done